mlrun/model_monitoring/db/tsdb/timescaledb/utils/timescaledb_query_builder.py

```diff
@@ -0,0 +1,257 @@
+            app_list = "', '".join(app_names)
+            return f"{mm_schemas.WriterEvent.APPLICATION_NAME} IN ('{app_list}')"
+        else:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid 'app_names' filter: must be a string or a list."
+            )
+
+    @staticmethod
+    def build_metrics_filter(
+        metrics: list[mm_schemas.ModelEndpointMonitoringMetric],
+    ) -> str:
+        """
+        Generate SQL filter for metrics.
+
+        :param metrics: List of ModelEndpointMonitoringMetric objects
+        :return: SQL WHERE clause fragment for metrics filtering
+        """
+        if not metrics:
+            raise mlrun.errors.MLRunInvalidArgumentError("Metrics list cannot be empty")
+
+        metric_names = [metric.name for metric in metrics]
+        if len(metric_names) == 1:
+            return f"{mm_schemas.MetricData.METRIC_NAME} = '{metric_names[0]}'"
+        metric_list = "', '".join(metric_names)
+        return f"{mm_schemas.MetricData.METRIC_NAME} IN ('{metric_list}')"
+
+    @staticmethod
+    def build_results_filter(
+        metrics: list[mm_schemas.ModelEndpointMonitoringMetric],
+    ) -> str:
+        """
+        Generate SQL filter for results using result_name column.
+        :param metrics: List of ModelEndpointMonitoringMetric objects
+        :return: SQL WHERE clause fragment for results filtering
+        """
+        if not metrics:
+            raise mlrun.errors.MLRunInvalidArgumentError("Metrics list cannot be empty")
+        metric_names = [metric.name for metric in metrics]
+        if len(metric_names) == 1:
+            return f"{mm_schemas.ResultData.RESULT_NAME} = '{metric_names[0]}'"
+        metric_list = "', '".join(metric_names)
+        return f"{mm_schemas.ResultData.RESULT_NAME} IN ('{metric_list}')"
+
+    @staticmethod
+    def build_metrics_filter_from_names(metric_names: list[str]) -> str:
+        """
+        Generate SQL filter for metrics by name.
+
+        :param metric_names: List of metric names
+        :return: SQL WHERE clause fragment for metrics filtering
+        """
+        if not metric_names:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Metric names list cannot be empty"
+            )
+
+        if len(metric_names) == 1:
+            return f"{mm_schemas.MetricData.METRIC_NAME} = '{metric_names[0]}'"
+        metric_list = "', '".join(metric_names)
+        return f"{mm_schemas.MetricData.METRIC_NAME} IN ('{metric_list}')"
+
+    @staticmethod
+    def combine_filters(filters: list[str], operator: str = "AND") -> Optional[str]:
+        """
+        Combine multiple filter conditions with the specified operator.
+
+        :param filters: List of filter condition strings
+        :param operator: SQL operator to use (AND/OR)
+        :return: Combined filter string or None if no filters
+        """
+        if valid_filters := [f for f in filters if f and f.strip()]:
```

Comment:
Not sure what `if f and f.strip()` is trying to achieve. Are we worried these strings will be empty? Even if that's the case `if f.strip()` would have the same effect.

Reply:
will fix